packages/spacecat-shared-http-utils/src/auth/handlers/scoped-api-key.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { hasText, isIsoDate } from '@adobe/spacecat-shared-utils';
+import AbstractHandler from './abstract.js';
+import { hashWithSHA256 } from '../generate-hash.js';
+import AuthInfo from '../auth-info.js';
+
+/**
+ * Handler to support API keys which include scope details. These API keys are stored in the data
+ * layer and require context.dataAccess in order to authenticate each request.
+ */
+export default class ScopedApiKeyHandler extends AbstractHandler {
+  constructor(log) {
+    super('scopedApiKey', log);
+  }
+
+  async checkAuth(request, context) {
+    const { dataAccess, pathInfo: { headers = {} } } = context;
+    if (!dataAccess) {
+      throw new Error('Data access is required');
+    }
+
+    const apiKeyFromHeader = headers['x-api-key'];
+    if (!hasText(apiKeyFromHeader)) {
+      return null;
+    }
+
+    // Keys are stored by their hash, so we need to hash the key to look it up
+    const hashedKey = hashWithSHA256(apiKeyFromHeader);
+    const apiKeyEntity = await dataAccess.getApiKeyByHashedKey(hashedKey);
+
+    if (!apiKeyEntity) {
```

Comment:
Do we need to check for empty braces `{}` as well? If the DB returns `{}` in case the hashedKey is not found?

Reply:
I had assumed `getApiKeyByHashedKey` would return `null` if the API key entity is not found, similar to:

```
export const getImportJobByID = async (dynamoClient, config, log, id) => {
  const item = await dynamoClient.getItem(
    config.tableNameImportJobs,
    { id },
  );
  return item ? ImportJobDto.fromDynamoItem(item) : null;
};
```

.. but see that this is not the case for `getApiKeyByHashedKey`. I think this should be updated to return null when not found for consistency — WDYT @swetabar?